### PR TITLE
chore(build): install clang-format/tidy in dockerfile, use proper ver…

### PR DESCRIPTION
**Pull request by JakubFornadel**
## Purpose

We did not installed properly clang-format and clang-tidy in docker, which were therefore not called during PR's so we merged changes into develop without proper formatting...

- updated to clang version 12
- updated doc/building 
- proper detection of clang-format and clang-tidy in cmake

## For reviewers

- conan with version >= 1.36 must be installed
- clang-12 must be installed -> see updated doc/building

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -1,3 +1,200 @@
...
+BreakBeforeConceptDeclarations: true
+BreakBeforeBraces: Attach
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: BeforeColon
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeColon
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     80
```
Comment (rattrap) [inline .clang-format:60]: ColumnLimit should be 120

File: Dockerfile [hunk 2]
```diff
@@ -28,7 +29,8 @@ RUN apt-get update \
 # Install LLVM
 RUN curl -SL -o llvm.sh https://apt.llvm.org/llvm.sh && \
     chmod +x llvm.sh && \
-    ./llvm.sh 10 && \
+    ./llvm.sh $LLVM_VERSION && \
+    apt-get install -y clang-format-$LLVM_VERSION clang-tidy-$LLVM_VERSION && \
```
Comment (rattrap) [inline Dockerfile:33]: Move this to the "# Install standard tools" section

File: CMakeModules/clang_tidy.cmake [hunk 3]
```diff
@@ -1,12 +1,18 @@
 # additional target to perform clang-tidy run, requires clang-tidy
 
+message(STATUS "clang-tidy LLVM_VERSION: ${LLVM_VERSION}")
+
 if (APPLE)
```
Comment (rattrap) [inline CMakeModules/clang_tidy.cmake:5]: No need to have this condition for APPLE. We can just keep the "else" find

**Review by rattrap — CHANGES_REQUESTED** [review]

File: .clang-format [hunk 4]
```diff
@@ -1,3 +1,200 @@
-Language: Cpp
-BasedOnStyle: Google
-ColumnLimit: 120
+---
+Language:        Cpp
+# BasedOnStyle:  Google
```
Comment (gaoqi8649) [inline .clang-format:3]: Why not just use Google style?

File: src/util/timer.hpp [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * const auto start_time = startTimer();
+ *
+ * ... some code that should be measured in time
+ *
+ * auto duration_us = stopTimer(start_time);
+ * std::cout << "duration: " << duration_us << " [ms]";
+ *
+ */
+
+}  // namespace taraxa::util
```
Comment (gaoqi8649) [inline src/util/timer.hpp:38]: Is this for test only? I don't find where dose use it

**Comment by JakubFornadel** [comment]
> Please also build at MacOS in CircleCI, make sure that also works for Mac. Thanks!

Agree, not ewlwvant comment for any PR though. This is rather a request for @agrebin @rattrap

**Comment by JakubFornadel** [comment]
> Don't forget to change add LLVM_VERSION here also:
> 
> ```
> RUN conan remote add -f bincrafters "https://api.bintray.com/conan/bincrafters/public-conan" && \
>     conan profile new clang --detect && \
>     conan profile update settings.compiler=clang clang && \
>     conan profile update settings.compiler.version=10 clang && \
>     conan profile update settings.compiler.libcxx=libstdc++11 clang && \
>     conan profile update env.CC=clang-10 clang && \
>     conan profile update env.CXX=clang++-10 clang && \
>     conan install -if $BUILD_OUTPUT_DIR --build missing -s build_type=Debug -pr=clang .
> ```

done

**Review by rattrap — APPROVED** [review]

**Review by justinsnapp — APPROVED** [review]

**Review by gaoqi8649 — APPROVED** [review]

**Pull request merged**